### PR TITLE
TT-191: Support other reference formats

### PR DESCRIPTION
**Pull request by vkarthim**
* Added OnBlur event to validate Reference format
* Added Regex function to validate the Reference format

### CODE REVIEW

File: src/portable/ReactUi/react/src/components/TaskDetails.tsx [hunk 1]
```diff
@@ -100,9 +103,9 @@ class TaskDetails extends React.Component<IProps, typeof initialState> {
                             <div className="taskItemContent">
                                 <div className="firstLine">
                                     <span className="displayReference">{reference}</span>
-                                    <span className={"totalTime" + (fileName !== ""? "": " hide")}>
-                                        <Duration seconds={this.duration()} direction={direction} />
-                                    </span>
+                                    <span className={"totalTime" + (fileName !== "" ? "" : " hide")}>
+                                        {durationWraper}
```
Comment (gtryus) [inline src/portable/ReactUi/react/src/components/TaskDetails.tsx]: This change seems to be included in the wrong commit. I think this change is related to TT-160

File: src/portable/ReactUi/react/src/components/TaskDetails.tsx [hunk 2]
```diff
@@ -123,6 +126,11 @@ class TaskDetails extends React.Component<IProps, typeof initialState> {
         )
     }
 
+    private validateReference(ref: string) {
+        const refExpr = /^\w{1,3}\s{1}\d{1,3}(\.|:){1}\d{1,3}(-|=)\d{1,3}$/;
+        this.setState({message: refExpr.test(ref)? "" : "Reference format: LUK 1:1-4 or LUK 1.1=4"});
```
Comment (gtryus) [inline src/portable/ReactUi/react/src/components/TaskDetails.tsx]: The message should be set up as a string that can be localized.

Comment (vkarthim) [reply src/portable/ReactUi/react/src/components/TaskDetails.tsx]: Ok

**Comment by gtryus** [comment]
Merged with changes as #115

**Pull request closed without merging**